bench/utils/__init__.py

```diff
@@ -0,0 +1,85 @@
+import struct
+import sys
+import time
+import random
+
+_RANDOM = random.Random(0)
+_SECTIONS = 10
+
+
+class _BaseBenchmark(object):
+    def __init__(self, name, ticks):
+        self.name = name
+        self.start_time = 0
+        self.end_time = 0
+        self.duration = 0
+        self.variance = 0.0
+        self._sections = []
+        self._watch = 0
+        self._ticks = ticks
+        self._ticked = 0
+
+    def dot(self):
+        sys.stdout.write(".")
+        sys.stdout.flush()
+
+    def start_watch(self):
+        self._watch = time.time()
+
+    def stop_watch(self):
+        stop = time.time()
```

Comment:
So `time.time()` is potentially a bit tricky. Ideally we'd use [this](https://pypi.python.org/pypi/Monotime), or something equivalent.